vignettes/FDZ_Data_Cleaning.Rmd

```diff
@@ -134,11 +116,26 @@ dat5 <- recodeGADS(dat5, varName = "age_FDZ", oldValues = c(13, 14, 15, 17),
                   newValues = c(13, 13, 15, 15))
 dat5 <- changeValLabels(dat5, varName = "age_FDZ", value = c(13, 15), valLabel = c("13-14", "15-17"))
 ```
+You can also perform grouping for multiple variables using a loop. For this we give the variables a temporary name consisting of the variable name and the suffix "_FDZ". `varname` stands here for each individual variable, which was summarized before in `gr_var`. The individual steps (`cloneVariable()`, `recodeGADS()`,`changeValLabels()`) are therefore performed for each of the variables in turn.
+```{r recodeGADS_loop}
+dat6 <- dat5
+gr_var <- c("grade_math", "grade_germ", "grade_eng")
+for (varname in gr_var) {
+  tempnewname <- paste0(varname, "_FDZ")
+  dat6 <- cloneVariable(dat6, varName = varname,
+                        new_varName = tempnewname, label_suffix = " (Zur Anonymisierung aggregiert (FDZ))")
+  dat6 <- recodeGADS(dat6, varName = tempnewname,
+                      oldValues = c(1:6),
+                      newValues = c(rep(1,4),rep(2,2)))
+  dat6 <- changeValLabels(dat6, varName = tempnewname, value = c(1, 2),
+                           valLabel = c("passed", "not passed"))
+}
+```
 
 ## Change variable and value labels
 To change variable or value labels, the functions `ChangeVarLabels()` and `ChangeValLabels()` are helpful. In `ChangeVarLabels()` the variable to be changed is selected under `varName` and with `varLabel` the corresponding label is assigned. `ChangeValLabels()` works the same way, except that the corresponding value is selected, too.
 ```{r changeVarLabels}
-dat6 <- changeVarLabels(dat5, varName = "ID_name", varLabel = "Vorname")
+dat6 <- changeVarLabels(dat6, varName = "ID_name", varLabel = "Vorname")
```

Comment:
Maybe adapt order. Introduce `changeValLabels()` before using it in the loop?

Reply:
good idea